fix: [오번역수정] Part 1 - 2.15 [함수] 매개변수 설명 수정 요청 (issue  #1444)

boyeonihn opened the pull request:
# Pull Request 체크리스트
## TODO
- [X] [번역 규칙을 확인하셨나요?](https://github.com/javascript-tutorial/ko.javascript.info#%EB%B2%88%EC%97%AD-%EA%B7%9C%EC%B9%99)
  - [X] 줄 바꿈과 단락을 '원문과 동일하게' 유지하셨나요?
  - [X] [맞춤법 검사기](http://speller.cs.pusan.ac.kr/)로 맞춤법을 확인하셨나요?
  - [X] 마크다운 문법에 사용되는 공백(스페이스), 큰따옴표("), 작은따옴표('), 대시(-), 백틱(`) 등의 특수문자는 그대로 두셨나요?
- [X] [로컬 서버 세팅](https://github.com/javascript-tutorial/ko.javascript.info/wiki/%EB%A1%9C%EC%BB%AC-%EC%84%9C%EB%B2%84-%EC%84%B8%ED%8C%85%ED%95%98%EA%B8%B0) 후 최종 결과물을 확인해 보셨나요?
- [X] PR 하나엔 번역문 하나만 넣으셨나요?
- [X] 의미 있는 커밋 메시지를 작성하셨나요?
  - 예시
    - [프락시] 번역
    - [프락시] 과제 번역
    - [if문과 조건부 연산자 '?'] 리뷰
    - [주석] 2차 리뷰
    - [Date 객체와 날짜] 번역

leviding committed:
Update article.md

dylan-workhub committed:
Fix plural typo line 414

Fixed a small typo with pluralization.

iliakan committed:
function parameters vs arguments

Yoona8 committed:
Typo in Ch 15 Functions

iliakan committed:
minor fixes

Rnbsov committed:
missing a dot

iliakan committed:
minor fixes

iliakan committed:
minor fixes

iliakan committed:
minor fixes

bhushan committed:
Update article.md

Violet-Bora-Lee commented:
혹시 영어 변경분에 대응하는 커밋 체리픽 해서 넣어주실수 있으실까요?
영어 원문을 한국어 원문과 머지 할 때 충돌 방지를 위해서 필요한 작업이긴하거등요.
시도해보시고 혹시 안된다면 알려주세요. 그냥 머지 하도록 하겠습니다~

boyeonihn commented:
보라님, 제가 체리픽에 대해서 읽어보긴 했는데 아직 원하시는 게 무엇인지 이해가 안 되어서요! (이래서 이론이 있어도 실천이 없으면 안되나봐요 🥲)
영어 변경분에 대응하는 새 단락의 한국 번역을 따로 커밋해서 PR해달라는 말씀인가요? 

1:1로 설명하시는 게 시간이 너무 오래 걸리실 거 같으면 멘토님께서 머지해주셔도 되어요. 제가 따로 조금 더 cherry pick에 관해 공부를 하곘습니다. 감사합니다!

boyeonihn commented:
> 이 작업은 그럼 원격저장소 관련해서 강의 한번 한다음에, 같이 사례로 다뤄봐도 좋을것같아요~ 그때 한번 같이 이야기 나눠보도록 해요 :)

앗 그래주시면 저도 다른 멘티분들도 정말 유익한 시간 될 것 같아요, 감사합니다 보라님!

iliakan committed:
closes #3111

tangjm committed:
Reword sentence on line 268

boyeonihn committed:
fix: [오번역수정] Part 1 - 2.15 [함수] 매개변수 설명 수정 (#1444)

영문 article.md에 대한 커밋을 cherry-pick를 하다가 매개변수 및 매개변수 기본값에 대한 설명이 추가되었음을 확인하고 번역 누락을 수정하기 위해 커밋 생성.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/javascript-tutorial/ko.javascript.info?pullRequest=1477) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/javascript-tutorial/ko.javascript.info?pullRequest=1477) before we can accept your contribution.<br/>**3** out of **8** committers have signed the CLA.<br/><br/>:white_check_mark: leviding<br/>:white_check_mark: iliakan<br/>:white_check_mark: boyeonihn<br/>:x: dylan-workhub<br/>:x: Yoona8<br/>:x: Rnbsov<br/>:x: tangjm<br/>:x: bhushan<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/javascript-tutorial/ko.javascript.info?pullRequest=1477) it.</sub>

boyeonihn commented:
보라님 제가 영문 레포에서 2.15 함수 챕터에 대한 커밋만 cherry-pick를 하고 conflict 수정을 하긴 했는데, 제대로 했는지 잘 모르겠네요 🥲

하지만 cherry-pick를 하다가 영문 아티클에 새로 추가된 섹션이 몇 개 더 있는 걸 발견하고 그것에 대한 번역도 포함했습니다. 시간되실 때 확인 부탁드려요. 감사해요!

boyeonihn commented:
/done

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -211,20 +226,58 @@ function showMessage(from, text = anotherFunction()) {
 ### 매개변수 기본값을 설정할 수 있는 또 다른 방법
 
-가끔은 함수 선언부에서 매개변수 기본값을 설정하는 것 대신 함수가 실행되는 도중에 기본값을 설정하는 게 논리에 맞는 경우가 생기기도 합니다. 
+가끔은 함수 선언할 때가 아닌 함수 선언 후 이후 단계에서 매개변수 기본값을 설정하는 것이 적절한 경우도 있습니다. 
```
여기가 좀 부자연스러운데, 

```suggestion
가끔은 함수를 선언할 때가 아닌 함수 선언 후 매개변수 기본값을 설정하는 것이 적절한 경우도 있습니다. 
```
이렇게 하면 어떨까요?

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -211,20 +226,58 @@ function showMessage(from, text = anotherFunction()) {
-  if (text === undefined) {
-    text = '빈 문자열';
+  if (text === undefined) { // 매개변수가 생략되었다면 
+    text = '빈 문자열'; 
```
불필요한 공백이 추가된 이유가있을까요?

boyeonihn replied:
에러입니다, 수정하겠습니다!

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
````diff
@@ -247,7 +300,7 @@ function showMessage(text) {
 
 ```js run
-// 매개변수 'count'가 넘어오지 않으면 'unknown'을 출력해주는 함수
+// 매개변수 'count'가 undefined 또는 null 값이면 'unknown'을 출력해주는 함수
````
```suggestion
// 매개변수 'count'가 undefined 또는 null이면 'unknown'을 출력해주는 함수
```

boyeonihn commented:
보라님 수정할 부분을 짚어주셔서 감사합니다! 세 부분에 대한 변경사항을 기존 commit에 추가하고 푸시했습니다 :) 감사합니다!

boyeonihn commented:
/done

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -137,26 +137,23 @@ alert( userName ); // 함수는 외부 변수에 접근하지 않습니다. 따
 ## 매개변수
 
-매개변수(parameter)를 이용하면 임의의 데이터를 함수 안에 전달할 수 있습니다. 매개변수는 *인수(argument)* 라고 불리기도 합니다(매개변수와 인수는 엄밀히 같진 않지만, 튜토리얼 원문을 토대로 번역하였습니다 - 옮긴이).
+매개변수(parameter)를 이용하면 임의의 데이터를 함수 안에 전달할 수 있습니다. 매개변수는 *인자* 라고 불리기도 합니다.
```
인자는 처음 등장하는 단어이므로 한-영 병기 표시 부탁드려도될까요?

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
````diff
@@ -175,19 +172,30 @@ showMessage(from, "Hello"); // *Ann*: Hello
 alert( from ); // Ann
 ```
 
+함수의 매개변수에 전달된 값을 *인수*라고 부르기도 합니다. 
````
인수 역시 본 글에서 처음 등장한 단어라면 한-영 병기 표시부탁드립니다.

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -175,19 +172,30 @@ showMessage(from, "Hello"); // *Ann*: Hello
+
+더 정확한 이해를 돕기 위해 용어를 다시 한번 정리해볼까요?  
+
+- 매개변수(parameter)는 함수 선언 방식 괄호 사이에 있는 변수입니다(선언 시 용어).
+- 인수(argument)는 함수를 호출할 때 매개변수에 전달되는 값입니다(호출 시 용어).
```
매개변수는 이미 위쪽에서 한-영 병기하였으므로 병기처리 안해도 될것 같습니다.

Violet-Bora-Lee replied:
선언 시 용어 <-- 이 부분이 잘 이해가 안가는데요... 선언 시 쓰이는 용어라고 이해해야 할까요? 그렇다면 '선언 시 쓰이는 용어'라고 해주시길 부탁드려요.

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -175,19 +172,30 @@ showMessage(from, "Hello"); // *Ann*: Hello
+- 매개변수(parameter)는 함수 선언 방식 괄호 사이에 있는 변수입니다(선언 시 용어).
+- 인수(argument)는 함수를 호출할 때 매개변수에 전달되는 값입니다(호출 시 용어).
+
+즉, 매개변수를 나열해 함수를 선언한 다음 인수를 전달해 함수를 호출합니다.
```
즉, 함수 선언시 매개변수를 나열하게 되고, 함수를 호출할 땐 인수를 전달해 호출합니다.

이렇게 하는건 어떨까요?

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -175,19 +172,30 @@ showMessage(from, "Hello"); // *Ann*: Hello
+
+즉, 매개변수를 나열해 함수를 선언한 다음 인수를 전달해 함수를 호출합니다.
+
+위의 예에서 함수 `showMessage`는 `from`, `text`라는 두 개의 매개변수로 선언된 후 `from`, `Hello`의 값을 가진 두 개의 인수로 호출되었습니다.
```
적, 의, 것 같은 단어는 되도록이면 쓰지 않는게 좋습니다.(문서에 언급되어있는 부분이니 한번 확인 부탁드려요)

Violet-Bora-Lee replied:
```suggestion
위 예에서 함수 `showMessage`는 `from`과 `text`라는 두 매개변수를 사용해 선언되었고, 그 후 호출시엔 `from`, `Hello`라는 두 인수를 사용해 호출되었습니다.
```

Violet-Bora-Lee replied:
제안 드리는 문장을 적어보았는데요, 맞춤범 검사기는 안돌려보았으니 한번 보시고 나은 문장을 선택해 주시면 좋을듯합니다.

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -175,19 +172,30 @@ showMessage(from, "Hello"); // *Ann*: Hello
+이렇게 코드를 작성해도 에러가 발생하지 않습니다. 두 번째 매개변수에 값을 전달하지 않았기 때문에 `text`엔 `undefined`가 할당될 뿐입니다. 따라서 에러 없이 `"Ann: undefined"`가 출력됩니다.
 
-매개변수에 값을 전달하지 않아도 그 값이 `undefined`가 되지 않게 하려면 '기본값(default value)'을 설정해주면 됩니다. 매개변수 오른쪽에 `=`을 붙이고 `undefined` 대신 설정하고자 하는 기본값을 써주면 되죠.   
+매개변수에 값을 전달하지 않아도 그 값이 `undefined`가 되지 않게 하려면 함수 선언할 때 `=`를 사용해 '기본값(default value)'을 설정해주면 됩니다. 
```
```suggestion
매개변수에 값을 전달하지 않아도 그 값이 `undefined`가 되지 않게 하려면 함수를 선언할 때 `=`를 사용해 '기본값(default value)'을 설정해주면 됩니다. 
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
`````diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
+
+```
+
+````smart header="구식 자바스크립트의 매개변수 기본값 설정 방법"
`````
```suggestion
````smart header="구식 자바스크립트에서 매개변수 기본값 설정 방법"
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
`````diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
+```
+
+````smart header="구식 자바스크립트의 매개변수 기본값 설정 방법"
+몇 년 전만 해도 자바스크립트는 매개변수 기본값에 대한 구문을 지원하지 않았습니다. 그래서 매개변수 기본값을 설정하기 위해 다른 방법을 사용했습니다.
`````
```suggestion
몇 년 전만 해도 자바스크립트엔 매개변수 기본값 관련 구문이 없었습니다. 그래서 매개변수 기본값을 설정하려면 다른 방법을 사용해야만 했죠.
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
`````diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
+````smart header="구식 자바스크립트의 매개변수 기본값 설정 방법"
+몇 년 전만 해도 자바스크립트는 매개변수 기본값에 대한 구문을 지원하지 않았습니다. 그래서 매개변수 기본값을 설정하기 위해 다른 방법을 사용했습니다.
+
+요즘에도 오래된 스크립트에서 매개변수 기본값 설정을 위해 사용된 구식 코드를 접할 수 있습니다.
`````
```suggestion
요즘에도 오래된 스크립트를 보다 보면 매개변수 기본값 설정 관련 코드를 접할 수 있습니다.
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
+
+요즘에도 오래된 스크립트에서 매개변수 기본값 설정을 위해 사용된 구식 코드를 접할 수 있습니다.
+
+예를 들어 구식 코드에서는 매개변수 기본값을 설정하기 위해 먼저 매개변수의 값이 `undefined`와 일치하는지 명시적으로 확인하고, 일치한다면 기본값을 설정했습니다. 
```
```suggestion
구식 코드에서는 매개변수 기본값 설정을 위해 먼저 매개변수 값이 `undefined`인지 명시적으로 확인하고, 일치하는 경우엔 기본값을 설정합니다.
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
````diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
+}
+```
+
+또한 논리 연산자 `||`를 사용해 매개변수 기본값을 설정하기도 했죠.
````
```suggestion
이 방법 말고도 논리 연산자 `||`를 사용해 매개변수 기본값을 설정는 방법도 있습니다.
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
````diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
+
+```js
+function showMessage(from, text) {
+  // text의 값이 falsy하면 기본값이 할당됨
````
```suggestion
  // text의 값이 falsy면 기본값이 할당됨
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
````diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
+```js
+function showMessage(from, text) {
+  // text의 값이 falsy하면 기본값이 할당됨
+  // 이 코드는 text == ""가 매개변수 text에 값을 아예 전달하지 않는 것과 같다고 가정합니다. 
````
```suggestion
  // 이 방식은 text == ""일 경우, text에 값이 전달되지 않은것과 같다고 간주합니다.
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
 ### 매개변수 기본값을 설정할 수 있는 또 다른 방법
 
-가끔은 함수 선언부에서 매개변수 기본값을 설정하는 것 대신 함수가 실행되는 도중에 기본값을 설정하는 게 논리에 맞는 경우가 생기기도 합니다. 
+가끔은 함수를 선언할 때가 아닌 함수 선언 후 매개변수 기본값을 설정하는 것이 적절한 경우도 있습니다. 
```
```suggestion
가끔은 함수를 선언할 때가 아닌 함수 선언 후에 매개변수 기본값을 설정하는 것이 적절한 경우도 있습니다.
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
```diff
@@ -211,19 +226,57 @@ function showMessage(from, text = anotherFunction()) {
+가끔은 함수를 선언할 때가 아닌 함수 선언 후 매개변수 기본값을 설정하는 것이 적절한 경우도 있습니다. 
 
-이런 경우엔 일단 매개변수를 `undefined`와 비교하여 함수 호출 시 매개변수가 생략되었는지를 확인합니다.
+이런 경우엔 일단 매개변수를 `undefined`와 비교하여 함수 호출 시 매개변수가 전달되었는지를 확인합니다.
```
```suggestion
이런 경우엔 함수를 호출할 때 매개변수를 `undefined`와 비교하여 매개변수가 전달되었는지를 확인합니다.
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
````diff
@@ -237,7 +290,7 @@ showMessage(); // 빈 문자열
 
 ```js
-// 매개변수가 생략되었거나 빈 문자열("")이 넘어오면 변수에 '빈 문자열'이 할당됩니다.
+// 매개변수가 생략되었거나 빈 문자열("")이 넘어오면 변수에 '빈 문자열'이 할당됩니다. 
````
```suggestion
// 매개변수가 생략되었거나 빈 문자열("")이 넘어오면 변수에 '빈 문자열'이 할당됩니다.
```

Violet-Bora-Lee commented on 1-js/02-first-steps/15-function-basics/article.md:
````diff
@@ -247,7 +300,7 @@ function showMessage(text) {
 
 ```js run
-// 매개변수 'count'가 넘어오지 않으면 'unknown'을 출력해주는 함수
+// 매개변수 'count'가 undefined 또는 null이면 'unknown'을 출력해주는 함수
````
```suggestion
// 매개변수 'count'가 `undefined` 또는 `null`이면 'unknown'을 출력해주는 함수
```

Violet-Bora-Lee reviewed (changes requested):
잘 이해가 안가는 문장에 대하여 코멘트 달아보았습니다..
제가 번역한 부분에서도 지금 보니 이상한 부분이 있어서 수정해두었거든요.
작업하실때 같이 커밋부탁드려도될까요~? 🙏

boyeonihn commented:
/done
